src/libaktualizr/config/config.cc

```diff
@@ -276,21 +278,34 @@ void Config::postUpdateValues() {
 
   checkLegacyVersion();
   initLegacySecondaries();
+  LOG_TRACE << "Final configuration that will be used: \n" << (*this);
 }
 
-void Config::updateFromDirs() {
-  for (const auto& dir : config_dirs_) {
-    for (const auto& config : Utils::glob((dir / "*.conf").string())) {
-      updateFromToml(config);
+void Config::updateFromDirs(const std::vector<boost::filesystem::path>& configs) {
+  std::map<std::string, boost::filesystem::path> configs_map;
+  for (const auto& config : configs) {
+    if (boost::filesystem::is_directory(config)) {
+      for (const auto& config_file : Utils::glob((config / "*.conf").string())) {
+        configs_map[config_file.filename().string()] = config_file;
```

Comment:
Isn't the opposite of the desired ordering? The first config dir specified on the command line should take first precedence over anything specified after that. This ordering (unless I'm missing something) is the reverse of that. I think the `for (const auto& config : configs)` needs to be reversed.

Reply:
It is not reversed. First specified directory will be processed first.